apps/gdalmdimtranslate_lib.cpp

```diff
@@ -1802,135 +1919,58 @@ GDALMultiDimTranslate(const char *pszDest, GDALDatasetH hDstDS, int nSrcCount,
 GDALMultiDimTranslateOptions *GDALMultiDimTranslateOptionsNew(
     char **papszArgv, GDALMultiDimTranslateOptionsForBinary *psOptionsForBinary)
 {
-    GDALMultiDimTranslateOptions *psOptions = new GDALMultiDimTranslateOptions;
+
+    auto psOptions = std::make_unique<GDALMultiDimTranslateOptions>();
 
     /* -------------------------------------------------------------------- */
-    /*      Handle command line arguments.                                  */
+    /*      Parse arguments.                                                */
     /* -------------------------------------------------------------------- */
-    const int argc = CSLCount(papszArgv);
-    for (int i = 0; papszArgv != nullptr && i < argc; i++)
+    try
     {
-        if (i < argc - 1 &&
-            (EQUAL(papszArgv[i], "-of") || EQUAL(papszArgv[i], "-f")))
-        {
-            ++i;
-            psOptions->osFormat = papszArgv[i];
-        }
-
-        else if (EQUAL(papszArgv[i], "-q") || EQUAL(papszArgv[i], "-quiet"))
-        {
-            if (psOptionsForBinary)
-                psOptionsForBinary->bQuiet = TRUE;
-        }
+        auto argParser = GDALMultiDimTranslateAppOptionsGetParser(
+            psOptions.get(), psOptionsForBinary);
 
-        else if (EQUAL(papszArgv[i], "-strict"))
-        {
-            psOptions->bStrict = true;
-        }
+        argParser->parse_args_without_binary_name(papszArgv);
 
-        else if (i < argc - 1 && EQUAL(papszArgv[i], "-array"))
-        {
-            ++i;
-            psOptions->aosArraySpec.push_back(papszArgv[i]);
-        }
-        else if (i < argc - 1 && EQUAL(papszArgv[i], "-arrayoption"))
+        // Check for invalid options:
+        // -scaleaxes is not compatible with -array = "view"
+        // -subset is not compatible with -array = "view"
+        if (std::find(psOptions->aosArraySpec.cbegin(),
+                      psOptions->aosArraySpec.cend(),
+                      "view") != psOptions->aosArraySpec.cend())
         {
-            ++i;
-            psOptions->aosArrayOptions.AddString(papszArgv[i]);
-        }
-        else if (i < argc - 1 && EQUAL(papszArgv[i], "-group"))
-        {
-            ++i;
-            psOptions->aosGroup.push_back(papszArgv[i]);
-        }
-
-        else if (i < argc - 1 && EQUAL(papszArgv[i], "-subset"))
-        {
-            ++i;
-            psOptions->aosSubset.push_back(papszArgv[i]);
-        }
-
-        else if (i < argc - 1 && EQUAL(papszArgv[i], "-scaleaxes"))
-        {
-            ++i;
-            CPLStringList aosScaleFactors(
-                CSLTokenizeString2(papszArgv[i], ",", 0));
-            for (int j = 0; j < aosScaleFactors.size(); j++)
-            {
-                psOptions->aosScaleFactor.push_back(aosScaleFactors[j]);
```

Comment:
this way of parsing scaleaxes should be preserved for backward compatibility

Reply:
done